integrations/access/common/app.go

```diff
@@ -172,6 +188,148 @@ func (a *BaseApp) onWatcherEvent(ctx context.Context, event types.Event) error {
 	}
 }
 
+// accessListMonitorRun will monitor access lists and post review reminders.
+func (a *BaseApp) accessListMonitorRun(ctx context.Context) error {
+	process := lib.MustGetProcess(ctx)
+	ctx, cancel := context.WithCancel(ctx)
+
+	// On process termination, explicitly cancel the context
+	process.OnTerminate(func(ctx context.Context) error {
+		cancel()
+		return nil
+	})
+
+	s := gocron.NewScheduler(time.UTC).Cron(accessListReviewCron)
+	log := logger.Get(ctx)
+
+	log.Info("Access list monitor is running")
+
+	job, err := s.Do(func() {
+		log.Info("Looking for Access List Review reminders")
+
+		var nextToken string
+		var err error
+		for {
+			var accessLists []*accesslist.AccessList
+			accessLists, nextToken, err = a.accessLists.ListAccessLists(ctx, 0 /* default page size */, nextToken)
+			if err != nil {
+				log.Errorf("error listing access lists: %v", err)
+				return
+			}
+
+			for _, accessList := range accessLists {
+				if err := a.notifyForAccessListReviews(ctx, accessList); err != nil {
+					log.WithError(err).Warn("Error notifying for access list reviews")
+				}
+			}
+
+			if nextToken == "" {
+				break
+			}
+		}
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Start the scheduler.
+	s.StartAsync()
+	a.alMonitorJob.SetReady(true)
+
+	select {
+	case <-ctx.Done():
+		s.Stop()
+	case <-job.Context().Done():
+		return trace.Wrap(job.Error())
+	}
+
+	log.Info("Access list monitor is finished")
+
+	return nil
+}
+
+// notifyForAccessListReviews will notify if access list review dates are getting close. At the moment, this
+// only supports notifying owners.
+func (a *BaseApp) notifyForAccessListReviews(ctx context.Context, accessList *accesslist.AccessList) error {
+	log := logger.Get(ctx)
+	allRecipients := make(map[string]Recipient, len(accessList.Spec.Owners))
+
+	now := a.clock.Now()
+	// Find the current notification window.
+	notificationStart := accessList.Spec.Audit.NextAuditDate
+
+	// If the current time before the notification start time, skip notifications.
+	if now.Before(notificationStart) {
+		log.Infof("Access list %s is not ready for notifications, notifications start at %s", accessList.GetName(), notificationStart.Format(time.RFC3339))
+		return nil
+	}
+
+	// Get the owners from the bot as recipients.
+	for _, owner := range accessList.Spec.Owners {
+		recipient, err := a.bot.FetchRecipient(ctx, owner.Name)
+		if err != nil {
+			log.WithError(err).Debugf("error getting recipient %s", owner.Name)
+			continue
+		}
+		allRecipients[owner.Name] = *recipient
+	}
+
+	if len(allRecipients) == 0 {
+		return trace.NotFound("no recipients could be fetched")
+	}
+
+	// Try to create base notification data with a zero notification date. If these objects already
+	// exist, that's okay.
+	for _, recipient := range allRecipients {
+		owner := recipient.Name
+		_, err := a.accessListPluginData.Create(ctx, accessListOwnerKey(accessList.GetName(), owner), pd.AccessListNotificationData{
```

Comment:
Looks like `Create()` expects the name of the associated resource (access request up til now) instead of an arbitrary key. I understand we're doing `accessListName/recipient` in order to be able to mark whether each recipient has been notified. I am not sure what the implications of using an arbitrary key might be. We might need to use a single `PluginData` and have something like a `map[string]time.Time` with recipients if it is a hard requirement to have it associated to a specific resource.

Also not a big deal, but this will likely leave orphaned `PluginData` instances if a reviewer is removed from the access list, or if the access request is deleted, since we do not have a cleanup of associated plugin data like there exists for access requests:
https://github.com/gravitational/teleport/pull/33415/files#diff-50263df585c0f309db079502bf120dc0306cff4cfc34489309f2d3a795e4a269R152-R162